simln-lib/src/test_utils.rs

```diff
@@ -45,3 +54,169 @@ pub fn create_nodes(n: usize, node_capacity: u64) -> Vec<(NodeInfo, u64)> {
         })
         .collect()
 }
+
+mock! {
+    pub Generator {}
+
+    impl fmt::Display for Generator {
+        fn fmt<'a>(&self, f: &mut fmt::Formatter<'a>) -> fmt::Result;
+    }
+
+    impl PaymentGenerator for Generator {
+        fn payment_start(&self) -> Option<Duration>;
+        fn payment_count(&self) -> Option<u64>;
+        fn next_payment_wait(&self) -> Result<Duration, PaymentGenerationError>;
+        fn payment_amount(&self, destination_capacity: Option<u64>) -> Result<u64, PaymentGenerationError>;
+    }
+}
+
+mock! {
+    pub LightningNode {}
+    #[async_trait]
+    impl crate::LightningNode for LightningNode {
+        fn get_info(&self) -> &NodeInfo;
+        async fn get_network(&mut self) -> Result<bitcoin::Network, LightningError>;
+        async fn send_payment(
+                &mut self,
+                dest: bitcoin::secp256k1::PublicKey,
+                amount_msat: u64,
+            ) -> Result<lightning::ln::PaymentHash, LightningError>;
+        async fn track_payment(
+                &mut self,
+                hash: &lightning::ln::PaymentHash,
+                shutdown: triggered::Listener,
+            ) -> Result<crate::PaymentResult, LightningError>;
+        async fn get_node_info(&mut self, node_id: &PublicKey) -> Result<NodeInfo, LightningError>;
+        async fn list_channels(&mut self) -> Result<Vec<u64>, LightningError>;
+    }
+}
+
+/// Type alias for the result of setup_test_nodes.
+type TestNodesResult = (
+    Vec<NodeInfo>,
+    HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>>,
+);
+
+/// A builder for creating mock Lightning nodes for testing purposes.
+///
+/// This struct provides a flexible way to configure test nodes with features like
+/// keysend support and specific networks. It supports both full node setups (with
+/// node info and clients) and client-only setups for network testing.
+///
+/// # Examples
+///
+/// Using the builder for more control:
+/// let (nodes, clients) = LightningTestNodeBuilder::new(5)
+///     .with_keysend_nodes(vec![0, 2, 4])
+///     .build_full();
+///
+/// Building clients with specific networks:
+/// let clients = LightningTestNodeBuilder::new(3)
+///     .with_networks(vec![Network::Bitcoin, Network::Testnet, Network::Regtest])
+///     .build_clients_only();
+pub struct LightningTestNodeBuilder {
+    node_count: usize,              // Required - must be provided at creation.
+    initial_balance: u64,           // Always has a value (default: 100,000).
+    keysend_indices: Vec<usize>,    // Always a vector (default: Vec filled with 0..node_count).
+    networks: Option<Vec<Network>>, // Can be None (default) or Some(networks).
```

Comment:
I don't think all these comments are necessary (node_count and networks are obvious IMO, looks too chatpgt-ey), and they should be above the field rather than inline when they are.